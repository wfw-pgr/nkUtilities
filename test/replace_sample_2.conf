# this is a sample of variable definition
# -- below is definitions. -- #

# <define>   @coef1 = 1.0   # this is coef1
# <define>   @coef2 = 1000
# <define>   @coef3 = @coef1+@coef2

$ <define>   @coef1 = 3.0   $ this is coef1
$ <define>   @coef2 = 22
$ <define>   @coef3 = @coef1 + @coef2
$ <define>   @coef5 = @coef4 + @coef3
$ <define>   @coef4 = @coef3
$ <define>   @type  = type-A


# -- to disable the definition, use ## or ### etc. -- #
# -- program catch following expressions "#\s*@(variable's name)\s# -- #

set01	     0.1   0.2       @coef3
set02	     0.2   @coef1    @coef2
Set03	     0.1   0.2       0.4
Set04	     0.1   0.2       @coef2
ErrorPoint   @coef4
ErrorPoint   @coef5
type	     @type
val_frm_tbl  @val_frm_tbl

$ <include> filepath = test/include.conf

$ <postProcess> echo "This is an echo test."
$ <postProcess> echo "This is an echo test."
$ <postProcess> cp test/replace_sample.conf test/replace_copy.conf