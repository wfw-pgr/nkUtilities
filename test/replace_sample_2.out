# this is a sample of variable definition
# -- below is definitions. -- #

# <define>   0.0001911 = 1.0   # this is coef1
# <define>   0.2222 = 3.0e8
# <define>   0.2223911 = 0.0001911+0.2222

$ <define>   @coef1 = 0.0001911   $ this is coef1
$ <define>   @coef2 = 0.2222
$ <define>   @coef3 = @coef1 + @coef2
$ <define>   @coef5 = @coef4 + @coef3
$ <define>   @coef4 = @coef3
$ <define>   @type  = type-A


# -- to disable the definition, use ## or ### etc. -- #
# -- program catch following expressions "#\s*@(variable's name)\s# -- #

set01	     0.1   0.2       0.2223911
set02	     0.2   0.0001911    0.2222
Set03	     0.1   0.2       0.4
Set04	     0.1   0.2       0.2222
ErrorPoint   0.2223911
ErrorPoint   0.2223911 + 0.2223911
type	     type-A
val_frm_tbl  value_from_table

$ <include> filepath = test/include.conf

$ <postProcess> echo "This is an echo test."
$ <postProcess> echo "This is an echo test."
$ <postProcess> cp test/replace_sample.conf test/replace_copy.conf


$$ ==================================================================================== $$
$$ ===                                variables List                                === $$
$$ ==================================================================================== $$
$$
$$  name                           :   value                                             
$$ ---------------------------------------------------------------------------------------
$$                   @val_frm_tbl :                                   value_from_table
$$                         @coef1 :                                          0.0001911
$$                         @coef2 :                                             0.2222
$$                         @coef3 :                                          0.2223911
$$                         @coef5 :                                 @coef4 + 0.2223911
$$                         @coef4 :                                          0.2223911
$$                          @type :                                             type-A
$$ ---------------------------------------------------------------------------------------
